index.ts

```diff
@@ -0,0 +1,552 @@
+//      or for non-competitive reasons are offloaded. eg. learnset details are very
+//      large and not needed in many circumstances.
+//    - All other APIs are loaded together for a gen (+ mod) - it's rare a consumer
+//      cares *just* about Abilities or *just* Moves, and if they do and are in an
+//      environment where loading more is a concern they can implement their own
+//      splitting on top of the raw data files.
+//    - The aliases.json data is used internally (to implement getByID's) but is
+//      not exported.
+//    - The individual APIs (eg. Abilities) are not just Maps, as their lookup function
+//      is more sophisticated (handles aliases, caching, etc).
+//    - Internally, references can be resolved lazily through Javascript getters (though
+//      need to use `Object.defineProperty(Foo.prototype, ...)` to be able to make
+//      enumerable) to avoid object instantiation overhead and avoid creating objects for
+//      fields which are never read.
+//    - Map will need to be polyfilled or support for browsers before IE11 will need to
+//      be dropped.
+//
+//  *** TODO ***:
+//
+//   - 'Species'/'SpeciesDetails' names are problematic as they are plural
+//   - Demonstrate how PS can wrap the specific types and still extend them in OMs
+//   - All the fields in the JSON should be able to be `null` as a sentinel for removal,
+//     can implement this generically using higher order Typescript utility type
+//     functionality.
+//   - 'CommonFoo' and 'JSONFoo' naming could use bikeshedding!
+//   - Review fields to ensure optional fields are made required in the rich APIs where
+//     it makes sense to do so.
+//
+
+// core.ts
+
+type ID = '' | string & {__isID: true};
+toID(s: string): ID;
+
+type Generation = 1 | 2 | 3 | 4 | 5 | 6 | 7;
+// NOTE: CAP is not currently considered a mod but should be...
+type Mod = 'LGPE' | 'Stadium' | 'CAP' | 'SSB' | 'Next' | 'PiC' | 'Mix And Mega' | 'VGC 17';
+
+type WeatherName =
+  | 'Sand'
+  | 'Sun'
+  | 'Rain'
+  | 'Hail'
+  | 'Harsh Sunshine'
+  | 'Heavy Rain'
+  | 'Strong Winds';
+
+type StatusName = 'par' | 'psn' | 'frz' | 'slp' | 'brn' | 'tox';
+
+// data.ts
+
+type DataKind = 'Ability' | 'Move' | 'Item' | 'Species';
+type Nonstandard = 'Glitch' | 'Past' | 'Future' | 'CAP' | 'LGPE' | 'Pokestar' | 'Custom';
+
+interface JSONData extends Omit<Data, 'id' | 'kind' | 'num' | 'gen'> {}
+interface Data {
+  id: ID;
+  kind: DataKind;
+  name: string;
+  gen: Generation;
+  isUnreleased?: boolean;
+  isNonstandard?: Nonstandard;
+}
+
+// natures.ts
+
+type NatureName = 'Adamant' | 'Bashful' | 'Bold' | 'Brave' | 'Calm' | 'Careful' | 'Docile' | 'Gentle' |
+  'Hardy' | 'Hasty' | 'Impish' | 'Jolly' | 'Lax' | 'Lonely' | 'Mild' | 'Modest' | 'Naive' | 'Naughty' |
+  'Quiet' | 'Quirky' | 'Rash' | 'Relaxed' | 'Sassy' | 'Serious' | 'Timid';
+
+class Natures {
+  getByID(id: ID): Nature | undefined;
+  next(): {value: Nature, done: boolean};
```

Comment:
```suggestion
  [Symbol.iterator](): IterableIterator<Nature>;
```

(see the definition of [`Array`](https://github.com/microsoft/TypeScript/blob/6d969f729dc631ea94f246d68d2cde9fd52dfa40/lib/lib.es2015.iterable.d.ts#L60), and `IterableIterator`/`Iterator` above)

Reply:
thanks! will apply everywhere.